### DISCUSSION TITLE
output the number of particle when simulating

### DISCUSSION THREAD

**Discussion by BoyuanTang331** [InteractiveComputerGraphics/SPlisHSPlasH#289] [Q&A] [answered]
hi，

when I use custom_scense.py to simulate the movement of water, if I want to print the number of particle for the water, should I use fluid.numParticles() [https://github.com/InteractiveComputerGraphics/SPlisHSPlasH/blob/b7f8321bf999fed3bcd31d00a6070a23458e385a/pySPlisHSPlasH/examples/add_field.py#L50](url)   
or  fluid.numActiveParticles() [https://github.com/InteractiveComputerGraphics/SPlisHSPlasH/blob/b7f8321bf999fed3bcd31d00a6070a23458e385a/pySPlisHSPlasH/examples/add_field.py#L20](url)

if the water particle is moving, this two args are same, when the particle is staying still will these two args different？

![image](https://github.com/InteractiveComputerGraphics/SPlisHSPlasH/assets/117408630/3b0a46b0-225e-4374-979b-c1dc9abc2294)

which is exactly represent the numbers of water particle when simulation>

best regards

**Comment by janbender** [comment, marked as answer]
Use "numActiveParticles()". If there are emitters in the scene, we allocate already the memory of all particles that will be emitted. Then numParticles give you the number of all particles (active + allocated but not yet emitted). numActiveParticles() returns the number of particles which are currently active in the simulation.
